app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -87,6 +112,98 @@ def get_templates(inventory)
         end
       end
 
+      ## InventoryObject refresh methods
+
+      def merge_projects_into_namespaces_graph(inventory)
+        collection = @inv_collections[:container_projects]
+
+        inventory["project"].each do |data|
+          h = parse_project(data)
+          # TODO: Assumes full refresh, and running after get_namespaces_graph.
+          #   Would be a problem with partial refresh.
```

Comment:
Old refresh had these assumptions (order & full refresh) everywhere.
Graph refresh is pretty close to not having any :-), and there are concrete wins to making them independent.
But yeah, there is more to track for partial refresh, I'll make assertions not TODOs.